Source/CombatExtended/CombatExtended/Verbs/Verb_LaunchProjectileChangeAble.cs

```diff
@@ -0,0 +1,144 @@
+
+        public int burstCount;
+
+        public int uses;
+
+        public int shotTime;
+
+        public void LoadDataFromXmlCustom(XmlNode xmlRoot)
+        {
+            foreach (XmlNode node in xmlRoot.ChildNodes)
+            {
+                if (node.Name.ToLower() == "uses")
+                {
+                    uses = ParseHelper.ParseIntPermissive(node.InnerText);
+                }
+                if (node.Name.ToLower() == "burstcount")
+                {
+                    burstCount = ParseHelper.ParseIntPermissive(node.InnerText);
+                }
+                if (node.Name.ToLower() == "chancetouse")
+                {
+                    chanceToUse = ParseHelper.ParseFloat(node.InnerText);
+                }
+                if (node.Name.ToLower() == "shottime")
+                {
+                    shotTime = ParseHelper.ParseIntPermissive(node.InnerText);
+                }
+                if (node.Name.ToLower() == "projectile")
+                {
+                    DirectXmlCrossRefLoader.RegisterObjectWantsCrossRef(this, "projectile", node.InnerText, null, null);
+                }
+
+            }
+        }
+    }
+    public class ProjectileChangeExt : DefModExtension
+    {
+        public List<AdditionalWeapon> additionalEquipment;
+    }
+
+    public class Verb_LaunchProjectileChangeAble : Verb_ShootCE
+    {
+        public Dictionary<AdditionalWeapon, int> UseManager = null;
+
+        public ProjectileChangeExt ChangerExt => this.CasterPawn?.def.GetModExtension<ProjectileChangeExt>() ?? null;
+
+        public ThingDef ProjectileInt;
+
+        public bool fireswitch;
+
+        public Pair<AdditionalWeapon, int> burstSwitcherPair;
+
+        public override ThingDef Projectile
+        {
+            get
+            {
+                if (fireswitch)
+                {
+                    if (burstSwitcherPair != null)
+                    {
+                        if (burstSwitcherPair.second > 0)
+                        {
+                            return burstSwitcherPair.first.projectile;
+                        }
+                    }
+
+                    return ProjectileInt;
+                }
+                return base.Projectile;
+            }
+        }
+
+        public override bool TryCastShot()
+        {
+            fireswitch = false;
+
+            if (ChangerExt != null)
+            {
+                if (UseManager.NullOrEmpty())
+                {
+                    if (UseManager == null)
+                    {
+                        UseManager = new Dictionary<AdditionalWeapon, int>();
+                    }
+
+                    foreach (var weapon in ChangerExt.additionalEquipment)
+                    {
+                        UseManager.Add(weapon, weapon.uses);
+                    }
+                }
+
+                if (this.Bursting /*&& ChangerExt.additionalEquipment.Any(x => x.shotTime == this.burstShotsLeft)*/)

```

Comment:
Why is this partially commented out?

Reply:
the check didn't seem to be useful, will just delete it for cleanliness